src/games/strategy/triplea/util/TransportUtils.java

```diff
@@ -47,24 +48,12 @@
     final Map<Unit, Unit> mapping = new HashMap<Unit, Unit>();
     final IntegerMap<Unit> addedLoad = new IntegerMap<Unit>();
     for (final Unit unit : canBeTransported) {
-
-      // Find first transport that has enough capacity
-      final int cost = UnitAttachment.get((unit).getType()).getTransportCost();
-      Unit loadedTransport = null;
-      for (final Unit transport : canTransport) {
-        final int capacity = TransportTracker.getAvailableCapacity(transport) - addedLoad.getInt(transport);
-        if (capacity >= cost) {
-          addedLoad.add(transport, cost);
-          mapping.put(unit, transport);
-          loadedTransport = transport;
-          break; // skip to next unit after loaded
-        }
-      }
+      final Optional<Unit> transport = loadUnitIntoFirstAvailableTransport(unit, canTransport, mapping, addedLoad);
 
       // Move loaded transport to end of list
-      if (loadedTransport != null) {
-        canTransport.remove(loadedTransport);
-        canTransport.add(loadedTransport);
+      if (transport.isPresent()) {
```

Comment:
I suspect you can simplify, everything should be in memory, order nanosecond runtimes. This performance improvement strikes me each time.